src/main/fc/tasks.c

```diff
@@ -160,21 +160,51 @@ static void taskUpdateAccelerometer(timeUs_t currentTimeUs)
 
 static void taskUpdateRxMain(timeUs_t currentTimeUs)
 {
-    if (!processRx(currentTimeUs)) {
-        return;
-    }
+    static enum {
+        CHECK, PROCESS, MODES, UPDATE
+    } state = CHECK;
+
+    switch (state) {
+    default:
+    case CHECK:
+        ignoreTaskTime();
+        rxProcessPending(true);
+        state = PROCESS;
+        break;
+
+    case PROCESS:
+        ignoreTaskTime();
```

Comment:
This seems to be a different change that is pretty much independent of the RX processing change - this should potentially go into its own pull request with its own explanation.

Reply:
This is a mechanism to have the scheduler ignore short task executions and to only count the longest state execution for the purposes of estimating task times. If this isn't done the scheduler will underestimate the amount of time required and not complete cycles in time, breaking determinism. Without this, this PR doesn't give the desired results. Of course this also has the undesirable (but preferable) side effect of making the scheduler reserve more time that is strictly necessary for states which execute quickly.

Reply:
Makes sense. But this also seems to be a drawback of this change, putting undue pressure on the scheduler. I wonder if there is a way to improve this?

Reply:
Unless we track task state duration in the scheduler (which would cost us in terms of memory) the only solutions would seem to be to code the state machine to be very balanced in time consumption per state or to provide a mechanism whereby a task with states can track their durations itself, and then pass a hint back to the scheduler about what's coming up on the next iteration.

Reply:
None of these seem to be attractive. I think for now we have to accept that this is a bit of a shim to give us the desired behaviour, and see if we can come up with a better way over time.

Reply:
There is a slight cost in terms of processor efficiency here, but to fix it will take code space and/or memory and those resources are more precious.